setup.py

```diff
@@ -9,11 +9,16 @@
 
 from distutils.command import build
 from distutils.spawn import spawn
+from distutils import sysconfig
 import sys
 import os
 
 import versioneer
 
+if sys.platform.startswith('linux'):
+    # Patch for #2555 to make wheels without libpython
+    sysconfig.get_config_vars()['Py_ENABLE_SHARED'] = 0
```

Comment:
Nice find!
I'm not sure what is more preferable, but without monkeypatching I think the following could also work:
```python
try:
    from setuptools.command import build_ext
except ImportError:
    from distutils.command import build_ext

class build_ext_no_shared_libpython(build_ext):
    def get_libraries(self, ext):
        if sys.platform.startswith('linux'):
            return ext.libraries
        return super(build_ext_no_shared_libpython, self).get_libraries(ext)

cmdclass['build_ext'] = build_ext_no_shared_libpython
```